.github/workflows/release.yml

```diff
@@ -5,6 +5,8 @@ on:
   push:
     branches:
       - master
+      # This will allow releasing beta versions from non-master releases.
+      # TODO: We need to release this version with other tags than beta. It can mess up the beta version as it will override it with older code.
```

Comment:
mmm good catch, but you should have addressed this before merging, as beta releases are automated based on commits to those branches